### PR TITLE
Save load update

### PR DESCRIPTION
**Pull request by jidaprano**
IMPORTANT: delete your last save file before testing this, some formatting things have changed and it will get confused if it tries to read an old save file

There's a lot of changes in this so it'll have to be pretty thoroughly tested; I've done lots of testing on my own machine but lmk if there's anything else anyone finds.

Everything implemented in this PR is from the comment in this work item: 
https://github.com/orgs/Angels-On-Rooftops/projects/1/views/1?sliceBy%5Bvalue%5D=jidaprano&pane=issue&itemId=80578926

REMINDER TO SELF: forgot ab this before PR, but check for non-null and unique item and checkpoint IDs before loading scene; it unfortunately seems like SerializeFields have to be nullable types and thus cannot be required at compile time but they should be checked ASAP on run

### CODE REVIEW

**Comment by prepsure** [comment]
works on my machine :]

File: Assets/Scripts/GameManagement/DataPersistence/DataPersistenceManager.cs [hunk 1]
```diff
@@ -92,9 +94,13 @@ public void LoadGame(CallbackContext context)
...
         }
-        catch (Exception e)
+        catch (FileNotFoundException)
+        {
+            Debug.Log("Save file not found, starting new game");
+        } catch(Exception e)
         {
             Debug.Log("Game not loaded: " + e.Message);
+            fileDataHandler.CloseFileLoad();
         }
```
Comment (cadenceblorbo) [inline Assets/Scripts/GameManagement/DataPersistence/DataPersistenceManager.cs:104]: good splitting of exceptions!

File: Assets/Scripts/GameManagement/DataPersistence/FileDataHandler.cs [hunk 2]
```diff
@@ -61,6 +62,7 @@ public void WriteObjectToJson(object data)
         {
             string dataToStore = JsonUtility.ToJson(data, true);
             fileWriter.WriteLine(dataToStore);
+            Debug.Log(dataToStore);
```
Comment (cadenceblorbo) [inline Assets/Scripts/GameManagement/DataPersistence/FileDataHandler.cs:65]: probably remove/comment this out before merging

File: Assets/Scripts/GameManagement/DataPersistence/FileDataHandler.cs [hunk 3]
```diff
@@ -75,6 +77,7 @@ public object ReadObjectFromJson(string jsonTag, Type returnType)
             fileReader.DiscardBufferedData();
             fileReader.BaseStream.Seek(0, SeekOrigin.Begin);
             string data = GetSingleJsonObjectString(jsonTag);
+            Debug.Log(data);
```
Comment (cadenceblorbo) [inline Assets/Scripts/GameManagement/DataPersistence/FileDataHandler.cs:80]: same as above

File: Assets/Scripts/GameManagement/DataPersistence/FileDataHandler.cs (hunk 2 above)
Comment (jidaprano) [reply Assets/Scripts/GameManagement/DataPersistence/FileDataHandler.cs:65]: This was something specific that Saran requested to allow for a little more transparency when saving/loading

**Comment by cadenceblorbo** [comment]
Things I noticed when testing:

- Saving collected items/equipment works!
- Removing key items already collected works!
- Saving checkpoints works!
- Equipment currently on the character is not saved, may be something to consider adding in the future.
- When saving from one scene and loading in another for the first time, the code logged a NullReferenceException on line 102. Nothing seemed to break, but might be good to keep an eye on/look into.

Overall, looks great!

**Comment by jidaprano** [comment]
Sounds good, gonna merge this as is and check out that nullref on 102 next sprint

**Pull request merged**